config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java

```diff
@@ -95,7 +95,7 @@ private BundleMetadata buildFullBundleMetadata(final List<Entity> entities, Proj
 
     private Collection<Metadata> getDefinedEntitiesMetadata(final List<Entity> definedEntities) {
         return definedEntities.stream().filter(FILTER_NON_ENV_ENTITIES_EXCLUDING_FOLDER)
-                .map(Entity::getMetadata).collect(Collectors.toList());
+                .map(e -> ((GatewayEntity)e.getEntity()).getMetadata()).collect(Collectors.toList());
```

Comment:
As we are taking metadata from actual gateway entity, do we need metadata implementation in Entity.java ?
Please remove if its not used

Reply:
Yes we need it for environmental entities